### PR TITLE
Explicitly require subscription id when `azd` config is not set

### PR DESCRIPTION
**Pull request by wbreza**
If azure subscription id is not set in either azd config, env var, or flag then calling `azd init --no-prompt` should fail.

### CODE REVIEW

**Comment by vhvb1989** [comment]
Is this a fix for: https://github.com/Azure/azure-dev/issues/1024 ?

File: cli/azd/pkg/account/manager.go [hunk 1]
```diff
@@ -62,27 +62,27 @@ func NewManager(configManager config.Manager, azCli azcli.AzCli) (*Manager, erro
 // Gets the default subscription for the logged in account.
 // 1. Returns AZD config defaults if exists
 // 2. Returns AZ CLI defaults if exists
```
Comment (vhvb1989) [inline cli/azd/pkg/account/manager.go:64]: is this still valid comment?

File: cli/azd/pkg/account/manager.go [hunk 2]
```diff
@@ -225,39 +225,22 @@ func (m *Manager) getAllSubscriptions(ctx context.Context) ([]*azcli.AzCliSubscr
...
 
-		defaultSubscription = &Subscription{
-			Id:       allSubscriptions[0].Id,
-			Name:     allSubscriptions[0].Name,
-			TenantId: allSubscriptions[0].TenantId,
-		}
+	subscriptionId := fmt.Sprint(configSubscriptionId)
+	subscription, err := m.azCli.GetAccount(ctx, subscriptionId)
+	if err != nil {
+		return nil, fmt.Errorf("failed retrieving subscription with ID '%s'. %w", subscriptionId, err)
```
Comment (vhvb1989) [inline cli/azd/pkg/account/manager.go:236]: We might be able to do more here.
This basically means that there's a value in `config.defaults` for the subscription and is not valid. So, we can ask the user if he wants to remove it (if running interactive).  Or we can just remove it from the config.default.
Or we can mention in the error message that the value is not valid or unknown and what to do to fix it (call azd config foo bar).

The current message might leave the user wondering `and what do I do now???`

**Comment by wbreza** [comment]
> Is this a fix for: #1024 ?

Yes, just linked them.

File: cli/azd/pkg/account/manager.go [hunk 3]
```diff
@@ -62,27 +62,27 @@ func NewManager(configManager config.Manager, azCli azcli.AzCli) (*Manager, erro
...
 	}
 
-	location, err := m.getDefaultLocation(ctx, subscription.Id)
+	var location *Location
 
-	// If we don't have a default location then either this is the first run experience
-	// or the location specified in configuration is invalid.
-	if err != nil {
+	if subscription == nil {
 		location = &defaultLocation
```
Comment (vhvb1989) [inline cli/azd/pkg/account/manager.go:74]: Can the user set a config.default location without setting a config.default subscription??

If yes, are we ignoring the config.default location set by the user here?  And always using `defaultLocation` whenever there's not a subscription default config?

File: cli/azd/pkg/account/manager.go (hunk 1 above)
Comment (wbreza) [reply cli/azd/pkg/account/manager.go:64]: No, will update the comments.

File: cli/azd/pkg/account/manager.go (hunk 3 above)
Comment (wbreza) [reply cli/azd/pkg/account/manager.go:74]: The locations are dependent on subscription.  The only ARM locations API always expects a subscription scope.
https://learn.microsoft.com/en-us/rest/api/resources/subscriptions/list-locations?tabs=HTTP

Comment (wbreza) [reply cli/azd/pkg/account/manager.go:74]: The user can set it manually in the config, but not through any of our flows.

File: cli/azd/pkg/account/manager.go (hunk 2 above)
Comment (wbreza) [reply cli/azd/pkg/account/manager.go]: Added more verbose error message for now.

File: cli/azd/pkg/account/manager.go (hunk 3 above)
Comment (vhvb1989) [reply cli/azd/pkg/account/manager.go:73]: > The user can set it manually in the config, but not through any of our flows.

When the user sets it manually, this line is ignoring that the value that the user set in the config. (when the subscription is not set).  We need to block users from setting the location manually without having a subscription.

Comment (wbreza) [reply cli/azd/pkg/account/manager.go:73]: I'm not going to add any special cases into `azd config set` at this time.

**Review by vhvb1989 — APPROVED** [review]

File: cli/azd/pkg/tools/azcli/azcli.go [hunk 4]
```diff
@@ -483,17 +464,6 @@ func (cli *azCli) createDefaultClientOptionsBuilder(ctx context.Context) *azsdk.
 		WithPerCallPolicy(azsdk.NewUserAgentPolicy(cli.UserAgent()))
 }
 
-// Azure Active Directory codes can be referenced via https://login.microsoftonline.com/error?code=<ERROR_CODE>,
-// where ERROR_CODE is the digits portion of an AAD error code. Example: AADSTS70043 has error code 70043
-// Additionally, https://learn.microsoft.com/azure/active-directory/develop/reference-aadsts-error-codes#aadsts-error-codes
-// is a helpful resource with a list of error codes and messages.
```
Comment (weikanglim) [inline cli/azd/pkg/tools/azcli/azcli.go:489]: Revert deletion. This is unrelated to the configuration message regex

Comment (weikanglim) [reply cli/azd/pkg/tools/azcli/azcli.go:489]: Nvm , we've removed all
